.github/workflows/build.yaml

```diff
@@ -89,39 +72,60 @@ jobs:
       uses: actions/checkout@v3
       with:
         fetch-depth: 1
-    - name: Installing dependencies
-      run: sh ./.github/scripts/install_deps.sh
-    - name: CMake
-      run: |
-        # Prepare build dir
-        echo "Will use $(nproc) cores."
-        mkdir build
-        pushd build
-        cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=ON ..
-        popd
-    - name: Restore cache
-      uses: actions/cache/restore@v3
-      with:
-        path: build/clang-tidy-cache
-        key: clang-tidy-cache-${{ github.sha }}
-        restore-keys: |
-          clang-tidy-cache-
-    - name: Check
-      # To run a single check, call:
-      # python3 ../utils/run-clang-tidy.py -checks=-*,my-check-prefix* | tee ../clang-tidy.log
-      run: |
-        # run-clang-tidy.py
-        pushd build
-        python3 ../utils/run-clang-tidy.py -cache | tee ../clang-tidy.log
-        popd
-    - name: Save cache
-      # We want to save the cache even if the report fails because of failed files
-      uses: actions/cache/save@v3
+    - name: Check for changed files
+      uses: dorny/paths-filter@v2
+      id: changed_files
       with:
-        path: build/clang-tidy-cache
-        key: clang-tidy-cache-${{ github.sha }}
-    - name: Report
-      run: utils/check_clang_tidy_results.py clang-tidy.log
+        # groups to check for changed files (as string, yaml anchors are resolved by the tool)
+        filters: |
+          clang_tidy:
+            - 'src/**'
+            - 'utils/check_clang_tidy_results.py'
+            - 'utils/run-clang-tidy.py'
+          compile: &compile
+            - 'CMakeLists.txt'
+            - 'cmake/**'
+            - 'src/**'
+          test_suite:
+            - *compile
+            - 'data/**'
```

Comment:
Dividing data further makes sense. But as `dorny/paths-filter` does not support excluding path, it will be tricky to do it properly (and not forget to list new paths when they are introduced).
I add a todo comment in the code (which prevents me from optimizing the line away).

The "testsuite-light" sounds nice. Grabbing a binary from master is not possible because there is no binary for ubuntu (on which the tests run).

Reply:
By the way, data/i18n/locales (containing only json files) and data/txts/developers.json should move to another place, they do not need not end up on the users computer (not trigger a build or test).

Reply:
I made some tries with skipping directories inside data/, it seems to work, see https://github.com/SimonHeimberg/widelands/pull/6#issuecomment-1839597867

Reply:
> The "testsuite-light" sounds nice. Grabbing a binary from master is not possible because there is no binary for ubuntu (on which the tests run).

It just needs to be cached for the 1 or 2 configs we want to run "testsuite-light" on.

> By the way, data/i18n/locales (containing only json files) and data/txts/developers.json should move to another place, they do not need not end up on the users computer (nor trigger a build or test).

I'm not quite sure about that. They're next to the files that are generated from them. There really isn't any other good place for them, and they're probably not worth a new top-level directory.

Reply:
But still not in this PR. :) To keep it reviewable, the changes to actual workflow logic should be kept to the absolute minimum required for separating the yaml files.

Reply:
Actually the pr title says the goal is to skip unnecessary jobs, not to separate the files. This point is already reached. And today's changes made separation better.